### PR TITLE
Jungle Turfs

### PR DESCRIPTION
**Pull request by imtakingabreakdontatme**
Adds dirt and water. These are mostly just placeholders so people can start mapping. I'm sure we can argue over the mechanics of water at some later point in time. Sprites are of course from Ausops.

ausops

### CODE REVIEW

File: code/game/turfs/simulated/floor/plating/dirt.dm [hunk 1]
```diff
@@ -0,0 +1,19 @@
+/turf/open/floor/plating/dirt
+	name = "dirt"
+	desc = "Upon closer examination, it's still dirt."
+	icon = 'icons/turf/floors.dmi'
+	icon_state = "dirt"
+	var/smooth_icon = 'icons/turf/floors/dirt.dmi'
+	smooth = SMOOTH_MORE|SMOOTH_BORDER
+	canSmoothWith
```
Comment (Iamgoofball) [inline code/game/turfs/simulated/floor/plating/dirt.dm]: ?????????

Comment (KorPhaeron) [inline code/game/turfs/simulated/floor/plating/dirt.dm]: look below

Comment (Iamgoofball) [inline code/game/turfs/simulated/floor/plating/dirt.dm]: yes but w h y

File: code/game/turfs/simulated/water.dm [hunk 2]
```diff
@@ -0,0 +1,13 @@
+/turf/open/water
```
Comment (lzimann) [inline code/game/turfs/simulated/water.dm]: Don't we already a water(that one from holodeck) tile?

Comment (KorPhaeron) [inline code/game/turfs/simulated/water.dm]: That one does literally nothing and has a different icon to fit with the beach anyway

Comment (lzimann) [inline code/game/turfs/simulated/water.dm]: We could make them the same path and add stuff/more depth to it. But whatever, that can easily be done in the future so you don't have to edit an infinite number of maps in this pr.

Comment (KorPhaeron) [inline code/game/turfs/simulated/water.dm]: One is a holoturf though

Comment (KorPhaeron) [inline code/game/turfs/simulated/water.dm]: Acidic(?) jungle water should probably function differently than shallow pool/beach water anyway

**Comment by OtherDalfite** [comment]
Fucking gorgeous. Thanks for making these!

**Comment by WJohn** [comment]
god damn ausops you da man

**Comment by PKPenguin321** [comment]
If you go above one of those large plants does it render over or under you

**Comment by KorPhaeron** [comment]
@PKPenguin321 

They don't exist in the code yet but it will be under probably

**Comment by GunHog** [comment]
Absolute beauty.

**Comment by Supermichael777** [comment]
not enough mushroom dominated poison jungle.

**Comment by Militaires** [comment]
@korphaeron should be over you tbh
or atleast some leaves are an overlay which render above you

File: code/game/turfs/simulated/floor/plating/dirt.dm (hunk 1 above)
Comment (RemieRichards) [inline code/game/turfs/simulated/floor/plating/dirt.dm]: Remove this line, it does nothing.

File: code/game/turfs/simulated/floor/plating/dirt.dm [hunk 3]
```diff
@@ -0,0 +1,19 @@
...
+	baseturf = /turf/open/chasm/straight_down/lava_land_surface
+	initial_gas_mix = "o2=14;n2=23;TEMP=300"
+	planetary_atmos = TRUE
+
+/turf/open/floor/plating/dirt/Initialize()
+	if (!canSmoothWith)
```
Comment (RemieRichards) [inline code/game/turfs/simulated/floor/plating/dirt.dm]: well this is always true, which makes this ```if(TRUE)```, so it's a no-op.

**Comment by FantasticFwoosh** [comment]
[Got you covered suggestionwise on how to make real river tiles](https://tgstation13.org/phpBB/viewtopic.php?f=9&t=10281), use @Xhuis 's water system, but have channelled tiles which work by double digging a tile to create a channel that allows the water already within it to flow but not spread to above tiles.

As to say you could divert or fill in the map's natural water river, run it into a geothermal/hydroelectronic system.

**Comment by Xhuis** [comment]
Point of note, my water tiles are not done. If it's a simplistic version like this that doesn't involve spreading and flooding than I can easily add it, though.

**Comment by Doppeltongue** [comment]
Kor, if you're going to add jungle water, you'd better add piranhas or something similar. They could just be a little shadow that appears in the water if necessary.

File: code/game/turfs/simulated/floor/plating/dirt.dm (hunk 3 above)
Comment (ChangelingRain) [inline code/game/turfs/simulated/floor/plating/dirt.dm]: subtypes?

Comment (ChangelingRain) [inline code/game/turfs/simulated/floor/plating/dirt.dm]: but yeah it's kind of pointless

**Comment by Iamgoofball** [comment]
NO

# add sponges

File: code/game/turfs/simulated/water.dm [hunk 4]
```diff
@@ -0,0 +1,13 @@
...
+	icon = 'icons/turf/floors.dmi'
+	icon_state = "riverwater"
+	baseturf = /turf/open/chasm/straight_down/lava_land_surface
+	initial_gas_mix = "o2=14;n2=23;TEMP=300"
+	planetary_atmos = TRUE
+	slowdown = 1
+	wet = TURF_WET_WATER
+
+/turf/open/water/HandleWet()
+	return //water does not dry off
```
Comment (AnturK) [inline code/game/turfs/simulated/water.dm]: Won't using lube on these make it perma lubed with this ?

Comment (RemieRichards) [inline code/game/turfs/simulated/water.dm]: yes

Comment (AnturK) [inline code/game/turfs/simulated/water.dm]: I kinda want current direction, so we can pollute these now.

Comment (RemieRichards) [inline code/game/turfs/simulated/water.dm]: just needs to be:

```/turf/open/water/HandleWet()
    if(wet == TURF_WET_WATER)
        return
    ...()
    MakeSlippery(TURF_WET_WATER) //rewet after ...() clears out lube/ice etc.
```

Comment (RemieRichards) [inline code/game/turfs/simulated/water.dm]: Current direction?

Comment (AnturK) [inline code/game/turfs/simulated/water.dm]: As in pour lube upstream to make whole river slippy.

Comment (RemieRichards) [inline code/game/turfs/simulated/water.dm]: that would be something for proper water turfs, not these stop-gaps.

Comment (AnturK) [inline code/game/turfs/simulated/water.dm]: Yep

File: code/game/turfs/simulated/floor/plating/dirt.dm [hunk 5]
```diff
@@ -5,15 +5,17 @@
...
 	smooth = SMOOTH_MORE|SMOOTH_BORDER
-	canSmoothWith
 	baseturf = /turf/open/chasm/straight_down/lava_land_surface
 	initial_gas_mix = "o2=14;n2=23;TEMP=300"
 	planetary_atmos = TRUE
 
 /turf/open/floor/plating/dirt/Initialize()
-	if (!canSmoothWith)
-		canSmoothWith = list(/turf/closed)
+	canSmoothWith = list(/turf/closed, /turf/open/floor/plating/dirt)
```
Comment (ChangelingRain) [inline code/game/turfs/simulated/floor/plating/dirt.dm]: why is this even in Initialize anyway

**Pull request merged**